### PR TITLE
`azurerm_management_group_subscription_association` - correctly mark as gone if not found during read

### LINKED ISSUE #23323: azurerm_management_group_subscription_association - could not read properties for Management Group
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.5.5

### AzureRM Provider Version

3.73.0

### Affected Resource(s)/Data Source(s)

azurerm_management_group_subscription_association

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "~>3.0"
    }
  }
}

provider "azurerm" {
  skip_provider_registration = true
  features {}
}

resource "azurerm_management_group" "example" {
  display_name = "example"
}

data "azurerm_billing_mca_account_scope" "example" {
  billing_account_name = "X"
  billing_profile_name = "X"
  invoice_section_name = "X"
}

resource "azurerm_subscription" "example" {
  subscription_name = "My Example MPA Subscription"
  billing_scope_id  = data.azurerm_billing_mca_account_scope.example.id
}

data "azurerm_subscription" "example" {
  subscription_id = azurerm_subscription.example.subscription_id
}

resource "azurerm_management_group_subscription_association" "example" {
  management_group_id = azurerm_management_group.example.id
  subscription_id     = data.azurerm_subscription.example.id
}
```

### Debug Output/Panic Output

```shell
Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: could not read properties for Management Group "X"
│
│   with azurerm_management_group_subscription_association.example,
│   on main.tf line 31, in resource "azurerm_management_group_subscription_association" "example":
│   31: resource "azurerm_management_group_subscription_association" "example" {
```

### Expected Behaviour

During the provider read of `azurerm_management_group_subscription_association `, if the expected Management Group has no children, it should be treated as an empty list instead of an error. Then because the expected Subscription is not a child of the Management Group, the Subscription should be re-associated to the proper Management Group. Error is [here](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/managementgroup/subscription_association_resource.go#L123) where instead the provider errors out if the Management Group's children are empty, whereas instead of an error the children list should just be set to an empty list. Then later found would be considered false and the Management Group ID set to empty forcing the re-association.

### Actual Behaviour

When a Terraform plan or apply is run after the Subscription is manually moved, outside of Terraform's control, and thus the Management Group has no other children Subscriptions, Terraform produces the following error message ``.

### Steps to Reproduce

1. Run above provided Terraform configuration `terraform apply`
3. Manually move the Subscription to a different Management Group (IE through the Portal)
4. Run `terraform plan` and error occures

### Important Factoids

_No response_

### References

https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/managementgroup/subscription_association_resource.go#L123

### PR DESCRIPTION
**Pull request by teowa**
fix #23323

### CODE REVIEW

File: internal/services/managementgroup/subscription_association_resource.go [hunk 1]
```diff
@@ -119,13 +119,9 @@ func resourceManagementGroupSubscriptionAssociationRead(d *pluginsdk.ResourceDat
...
-		if props.Children == nil {
-			return fmt.Errorf("could not read properties for Management Group %q", id.ManagementGroup)
-		}
-
-		for _, v := range *props.Children {
-			if v.Type == managementgroups.Type1Subscriptions {
-				if v.Name != nil && *v.Name == id.SubscriptionId {
+		if props.Children != nil {
+			for _, v := range *props.Children {
+				if v.Type == managementgroups.Type1Subscriptions && v.Name != nil && *v.Name == id.SubscriptionId {
```
Comment (jackofallops) [inline internal/services/managementgroup/subscription_association_resource.go:124]: We should probably make this a case insensitive check here?
```suggestion
				if v.Type == managementgroups.Type1Subscriptions && v.Name != nil && strings.EqualFold(*v.Name, id.SubscriptionId) {
```

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @teowa - Thanks for this, just one comment to look at then I think this will be good to go.

File: internal/services/managementgroup/subscription_association_resource.go (hunk 1 above)
Comment (teowa) [reply internal/services/managementgroup/subscription_association_resource.go]: fixed, thanks.

**Review by stephybun — APPROVED** [review]
Thanks @teowa LGTM 👍

**Pull request merged**